internal/providers/pluginfw/resources/qualitymonitor/resource_quality_monitor.go

```diff
@@ -0,0 +1,214 @@
+	resp.TypeName = "databricks_quality_monitor_pluginframework"
+}
+
+func (r *QualityMonitorResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Quality Monitor",
+		Attributes: tfschema.ResourceStructToSchemaMap(MonitorInfoExtended{}, func(c tfschema.CustomizableSchema) tfschema.CustomizableSchema {
+			c.SetRequired("assets_dir")
+			c.SetRequired("output_schema_name")
+			c.SetRequired("table_name")
+			c.SetReadOnly("monitor_version")
+			c.SetReadOnly("drift_metrics_table_name")
+			c.SetReadOnly("profile_metrics_table_name")
+			c.SetReadOnly("status")
+			c.SetReadOnly("dashboard_id")
+			c.SetReadOnly("schedule", "pause_status")
+			return c
+		}),
+	}
+}
+
+func (d *QualityMonitorResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	d.Client = pluginfwcommon.ConfigureResource(req, resp)
+}
+
+func (r *QualityMonitorResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	if diags.HasError() {
+		return
+	}
+	var monitorInfoTfSDK MonitorInfoExtended
+	diags = req.Plan.Get(ctx, &monitorInfoTfSDK)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var createMonitorGoSDK catalog.CreateMonitor
+	diags = converters.TfSdkToGoSdkStruct(ctx, monitorInfoTfSDK, &createMonitorGoSDK)
+	if diags.HasError() {
+		return
+	}
+	monitor, err := w.QualityMonitors.Create(ctx, createMonitorGoSDK)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get created monitor", err.Error())
+		return
+	}
+	resp.Diagnostics.Append(waitForMonitor(ctx, w, monitor)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var newMonitorInfoTfSDK MonitorInfoExtended
+	resp.Diagnostics.Append(converters.GoSdkToTfSdkStruct(ctx, monitor, &newMonitorInfoTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, newMonitorInfoTfSDK)...)
+}
+
+func (r *QualityMonitorResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	if diags.HasError() {
+		return
+	}
+
+	var getMonitor catalog_tf.GetQualityMonitorRequest
+	diags = req.State.GetAttribute(ctx, path.Root("table_name"), &getMonitor.TableName)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	endpoint, err := w.QualityMonitors.GetByTableName(ctx, getMonitor.TableName.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get monitor", err.Error())
+		return
+	}
+	var monitorInfoTfSDK MonitorInfoExtended
+	diags = converters.GoSdkToTfSdkStruct(ctx, endpoint, &monitorInfoTfSDK)
+	if diags.HasError() {
+		return
+	}
+	diags = resp.State.Set(ctx, monitorInfoTfSDK)
+	resp.Diagnostics.Append(diags...)
```

Comment:
Let's be consistent here:
```suggestion
	resp.Diagnostics.Append(resp.State.Set(ctx, monitorInfoTfSDK)...)
```